ci-operator/step-registry/ipi/conf/aws/user-permissions/ipi-conf-aws-user-permissions-commands.sh

```diff
@@ -0,0 +1,290 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+trap 'rm -rf /tmp/aws_cred_output /tmp/pull-secret /tmp/min_perms/' EXIT TERM INT
+
+USER_POLICY_FILENAME="aws-permissions-policy-creds.json"
+USER_POLICY_FILE="${SHARED_DIR}/${USER_POLICY_FILENAME}"
+
+if [[ "${AWS_INSTALL_USE_MINIMAL_PERMISSIONS}" != "yes" ]]; then
+	echo "Custom AWS user with minimal permissions is disabled. Using AWS user from cluster profile."
+	exit 0
+fi
+
+function install_jq() {
+	echo "Checking/installing jq..."
+	if ! command -v jq; then
+		curl -L https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64 -o /tmp/bin/jq && chmod +x /tmp/bin/jq
+	fi
+	echo "Installing jq done: $(which jq)"
+}
+install_jq
+
+RELEASE_IMAGE_INSTALL="${RELEASE_IMAGE_INITIAL:-}"
+if [[ -z "${RELEASE_IMAGE_INSTALL}" ]]; then
+	# If there is no initial release, we will be installing latest.
+	RELEASE_IMAGE_INSTALL="${RELEASE_IMAGE_LATEST:-}"
+fi
+cp ${CLUSTER_PROFILE_DIR}/pull-secret /tmp/pull-secret
+oc registry login --to /tmp/pull-secret
+ocp_version=$(oc adm release info --registry-config /tmp/pull-secret ${RELEASE_IMAGE_INSTALL} --output=json | jq -r '.metadata.version' | cut -d. -f 1,2)
+ocp_major_version=$(echo "${ocp_version}" | awk --field-separator=. '{print $1}')
+ocp_minor_version=$(echo "${ocp_version}" | awk --field-separator=. '{print $2}')
+rm /tmp/pull-secret
+
+export AWS_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred"
+
+if ((ocp_major_version < 4 || (ocp_major_version == 4 && ocp_minor_version < 18))); then
+	# There is no installer support for generating permissions prior to 4.18, so we generate one ourselves
+	PERMISION_LIST="${ARTIFACT_DIR}/permision_list.txt"
+
+	cat <<EOF >"${PERMISION_LIST}"
```

Comment:
Does this mean this PR is trying to merge the functionality of `aws-provision-iam-user-minimal-permission` here, right?